src/main/java/com/hierynomus/smbj/smb2/messages/SMB2IoctlRequest.java

```diff
@@ -59,15 +57,24 @@ protected void writeTo(SMBBuffer smbBuffer) {
         fileId.write(smbBuffer);  // FileId (16 bytes)
 
         int offset = SMB2Header.STRUCTURE_SIZE + 56;
-        smbBuffer.putUInt32(offset); // InputOffset (4 bytes)
-        smbBuffer.putUInt32(inputData.length); // InputCount (4 bytes)
-        smbBuffer.putUInt32(MAX_OUTPUT_BUFFER_LENGTH); // MaxInputResponse (4 bytes)
+        if (inputData.length > 0) {
+            smbBuffer.putUInt32(offset); // InputOffset (4 bytes)
+            smbBuffer.putUInt32(inputData.length); // InputCount (4 bytes)
+        } else {
+            smbBuffer.putUInt32(0); // InputOffset (4 bytes)
```

Comment:
The Spec says,

The following FSCTL requests do not provide an input buffer: 􏰀 FSCTL_PIPE_PEEK
􏰀 FSCTL_PIPE_WAIT
􏰀 FSCTL_PIPE_TRANSCEIVE
􏰀 FSCTL_SRV_ENUMERATE_SNAPSHOTS
􏰀 FSCTL_SRV_REQUEST_RESUME_KEY
􏰀 FSCTL_QUERY_NETWORK_INTERFACE_INFO

Not sure if we need to check that inputData is empty for these request and non empty for all other requests. If you strongly feel it should be there, I can add it.